src/hooks/fire/useLiveCollection.ts

```diff
@@ -35,7 +35,7 @@ export const useLiveCollection = <T extends object, ID extends string = string>(
     hasDeferred,
     constraints,
     filteredConstraints,
-  } = parseCollectionQueryOptions(options);
+  } = useMemo(() => parseCollectionQueryOptions(options), [options]);
```

Comment:
Interesting, I didn't think this could work considering `options` itself is an object and maybe it's fields would need direct reference.

Maybe it does not change much or anything if only `options` are passed into `useLiveCollection`, but that R&D would be for another PR.